src/pybind/mgr/dashboard/services/auth/oauth2.py

```diff
@@ -0,0 +1,151 @@
+import json
+from typing import Dict, List
+from urllib.parse import quote
+
+import cherrypy
+import requests
+
+from ... import mgr
+from ...services.auth import BaseAuth, SSOAuth, decode_jwt_segment
+from ...tools import prepare_url_prefix
+from ..access_control import Role, User, UserAlreadyExists
+
+
+class OAuth2(SSOAuth):
+    LOGIN_URL = 'auth/oauth2/login'
+    LOGOUT_URL = 'auth/oauth2/logout'
+    sso = True
+
+    class OAuth2Config(BaseAuth.Config):
+        pass
+
+    @staticmethod
+    def enabled():
+        return mgr.get_module_option('sso_oauth2')
+
+    def to_dict(self) -> 'BaseAuth.Config':
+        return self.OAuth2Config()
+
+    @classmethod
+    def from_dict(cls, s_dict: OAuth2Config) -> 'OAuth2':
+        # pylint: disable=unused-argument
+        return OAuth2()
```

Comment:
Same thing here: what's the sense of this? It's ignoring the input `s_dict`.

Reply:
Same thing as before, and it's reciving the `s_dict` because of the base class implementation (should I remove the argument from there?) and it's ignoring the `s_dict` because the OAuth2 does not require any config.

I left it as it is to keep consistency with the Saml2 and with the idea to, maybe in the future and if needed, add config parameters to the OAuth2 and unify the Saml2 and OAuth2 implementations